atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/ExpectationsTask.java

```diff
@@ -30,13 +33,23 @@ public ExpectationsTask(Set<ExpectationsAwareTransaction> transactions) {
         this.transactions = transactions;
     }
 
+    // todo(aalouane) test this class after impl is done
     @Override
     public void run() {
         try {
-            Set<ExpectationsViolation> violations = transactions.stream()
-                    .flatMap(transaction -> transaction.checkAndGetViolations().stream())
-                    .collect(Collectors.toSet());
-            updateMetrics(violations);
+
+            Set<TransactionViolationFlags> flagSet = transactions.stream()
+                    .map(ExpectationsAwareTransaction::checkAndGetViolations)
+                    .collect(ImmutableSet.toImmutableSet());
+
+            updateMetrics(ImmutableTransactionViolationFlags.builder()
+                    .ranForTooLong(flagSet.stream().anyMatch(TransactionViolationFlags::ranForTooLong))
+                    .readTooMuch(flagSet.stream().anyMatch(TransactionViolationFlags::readTooMuch))
+                    .readTooMuchInOneKvsCall(
+                            flagSet.stream().anyMatch(TransactionViolationFlags::readTooMuchInOneKvsCall))
+                    .queriedKvsTooMuch(flagSet.stream().anyMatch(TransactionViolationFlags::queriedKvsTooMuch))
+                    .build());
```

Comment:
Hesitating on whether to create a new file `ViolationFlags` where I implement a static method equivalent to the code here, i.e.
```
public static TransactionViolationFlags any(Iterable<TransactionViolationFlags> flagsIterable)
```